[release] update changelog (#1741)
ginkgo-project/ginkgo: CHANGELOG.md

Please visit our wiki [Changelog](https://github.com/ginkgo-project/ginkgo/wiki/Changelog) for unreleased changes.

## Version 1.9.0

The Ginkgo team is proud to announce the new Ginkgo minor release 1.9.0.
This release brings new features such as:

> - Supporting half precision (IEEE FP16). The type `gko::half` can now be selected in most instances as the value-type

Reviewer: ```suggestion
- Supporting half precision (IEEE FP16). The type `gko::half` can now be selected in most instances as the value type
```

> - Supporting half precision (IEEE FP16). The type `gko::half` can now be selected in most instances as the value type

Reviewer 2: For the highlights, I would have the same language as for the previous release. Adding an inflection is not necessary. Same for the other points below. (Providing -> Provide, Adding -> Add, Simplifying -> Simplify)
```suggestion
- Support for half precision (IEEE FP16). The type `gko::half` can now be selected in most instances as the value type
```

  of a matrix, solver, preconditioner, etc. If the selected backend supports FP16 as a native type, the native type is

>   used within the kernels, otherwise an overhead might occur. The new behavior is enabled by default, but it can be

Reviewer 2: ```suggestion
  used within the kernels, otherwise an overhead might be inccured. The new behavior is enabled by default, but it can be
```

Author: I think the first one is grammatically correct. IMO the second one is no improvement, so I will not add it.

  turned off during configuration.

> - Providing new implementations of the ILU and IC factorization for CUDA, HIP, Omp, and Reference backends. These are

Reviewer: ```suggestion
- Providing new implementations of the ILU and IC factorization for CUDA, HIP, OpenMP, and Reference backends. These are
```

  available in addition to the existing implementations based on the vendor libraries cuSPARSE and hipSPARSE.    

> - New (S)SOR and Gauß-Seidel preconditioners.

Reviewer 3: ```suggestion
- Adding New (S)SOR and Gauß-Seidel preconditioners.
```
nit: just make all start with Ving

> - Adding new (S)SOR and Gauß-Seidel preconditioners.

Reviewer 2: Probably better to use the standard spelling ? I guess Gauß is more accurate, but non-standard. In the PR list below it does say Gauss.
```suggestion
- Adding new (S)SOR and Gauss-Seidel preconditioners.
```

> - Simplifying the distributed matrix assembly by exchanging local rows between neighboring processes.

Reviewer 3: ```suggestion
- Simplifying the distributed matrix assembly by gathering local rows from neighboring processes.
```
maybe?

And more!

If you face an issue, please first check our [known issues page](https://github.com/ginkgo-project/ginkgo/wiki/Known-Issues) and the [open issues list](https://github.com/ginkgo-project/ginkgo/issues) and if you do not 
find a solution, feel free to [open a new issue](https://github.com/ginkgo-project/ginkgo/issues/new/choose) or ask a question using the [github discussions](https://github.com/ginkgo-project/ginkgo/discussions).

Supported systems and requirements:
+ For all platforms, CMake 3.16+
+ C++17 compliant compiler
+ Linux and macOS
  + GCC: 7.0+
  + clang: 5.0+
  + Intel compiler: 2019+
  + Apple Clang: 15.0 is tested. Earlier versions might also work.
  + NVHPC: 22.7+
  + Cray Compiler: 14.0.1+

>   + CUDA module: CMake 3.18+, and CUDA 11.0+ or NVHPC 22.7+, Compute Capability 5.3+

Reviewer 3: ```suggestion
  + CUDA module: CMake 3.18+, and CUDA 11.0+ or NVHPC 22.7+
```
after #1742 merge

  + HIP module: CMake 3.21+, and ROCm 4.5+
  + DPC++ module: Intel oneAPI 2023.1+ with oneMKL and oneDPL. Set the CXX compiler to `dpcpp` or `icpx`.
  + MPI: standard version 3.1+, ideally GPU Aware, for best performance
+ Windows
  + MinGW: GCC 7.0+
  + Microsoft Visual Studio: VS 2019+
  + CUDA module: CUDA 11.0+, Microsoft Visual Studio
  + OpenMP module: MinGW.

### Version support changes
+ Ginkgo now requires a compiler with C++ 17 support [#1603](https://github.com/ginkgo-project/ginkgo/pull/1603)

### Deprecations

> + The `Executor::run` overload without a name as first parameter has been deprecated [#1667](https://github.com/ginkgo-project/ginkgo/pull/1667)

Reviewer: ```suggestion
+ The `Executor::run` overload taking in multiple functions without a name as first parameter has been deprecated [#1667](https://github.com/ginkgo-project/ginkgo/pull/1667)
```

> ### Summary of previous deprecations

Reviewer 2: This and the deprecations header above are at the same heading level. Should this one be a part of the one above ?

Author: That's a good idea. I will make it a sub-heading.

+ Logger constructors and create functions with the `executor` parameter.
+ The virtual, protected, Dense functions `compute_norm1_impl`, `add_scaled_impl`, etc.
+ Logger events for solvers and criterion without the additional `implicit_tau_sq` parameter.
+ The global `gko::solver::default_krylov_dim`, use instead `gko::solver::gmres_default_krylov_dim`.
+ `array::get_num_elems()` has been renamed to `get_size()`
+ `matrix_data::ensure_row_major_order()` has been renamed to `sort_row_major()`
+ `device_matrix_data::get_num_elems()` has been renamed to `get_num_stored_elements()`
+ The CMake parameter `GINKGO_COMPILER_FLAGS` has been superseded by `CMAKE_CXX_FLAGS`, and `GINKGO_CUDA_COMPILER_FLAGS` has been superseded by `CMAKE_CUDA_FLAGS`
+ The `std::initializer_list` overloads of matrix `create` methods and constructors are deprecated in favor of explicit `array` parameters

### Added features
+ Add `Executor::get_description()` for textual representation of the device [#1615](https://github.com/ginkgo-project/ginkgo/pull/1615)
+ Add row and column scaling functionality to the distributed matrix [#1640](https://github.com/ginkgo-project/ginkgo/pull/1640)

> + Add `SolverProgress` logger printing out or storing to disk the individual scalars (and vectors) of an iterative solvers after each iteration [#1620](https://github.com/ginkgo-project/ginkgo/pull/1620)

Reviewer: ```suggestion
+ Add `SolverProgress` logger printing out or storing to disk the individual scalars (and vectors) of an iterative solver after each iteration [#1620](https://github.com/ginkgo-project/ginkgo/pull/1620)
```

+ Add new `ortho_method` parameter for GMRES, with classical Gram-Schmidt and classical Gram-Schmidt with re-orthogonalization options in addition to previously-available modified Gram-Schmidt [#1646](https://github.com/ginkgo-project/ginkgo/pull/1646)
+ Add file config support for Schwarz [#1658](https://github.com/ginkgo-project/ginkgo/pull/1658)
+ Add overload for `Executor::run` which accepts a name and a closure for the ReferenceExecutor as the first two arguments [#1667](https://github.com/ginkgo-project/ginkgo/pull/1667)
+ Add function to fill `device_matrix_data` with zeros [#1683](https://github.com/ginkgo-project/ginkgo/pull/1683)
+ Add (S)SOR and Gauss-Seidel preconditioner [#1633](https://github.com/ginkgo-project/ginkgo/pull/1633), [#1634](https://github.com/ginkgo-project/ginkgo/pull/1634)

> + Add support for additive `read_distributed` for the distriubted matrix [#1650](https://github.com/ginkgo-project/ginkgo/pull/1650)

Reviewer: ```suggestion
+ Add support for additive `read_distributed` for the distributed matrix [#1650](https://github.com/ginkgo-project/ginkgo/pull/1650)
```

+ Add Ginkgo's own ILU and IC implementation [#1684](https://github.com/ginkgo-project/ginkgo/pull/1684)

> + Add NVIDIA ada architecture [#1733](https://github.com/ginkgo-project/ginkgo/pull/1733)

Reviewer: ```suggestion
+ Add NVIDIA Ada architecture [#1733](https://github.com/ginkgo-project/ginkgo/pull/1733)
```

Ginkgo 1.0.0 is brought to you by:

**Karlsruhe Institute of Technology**, Germany
**Universitat Jaume I**, Spain
**University of Tennessee, Knoxville**, US

These universities, along with various project grants, supported the development team and provided resources needed for the development of Ginkgo.

Ginkgo 1.0.0 contains contributions from:

**Hartwig Anzt**, Karlsruhe Institute of Technology
**Yenchen Chen**, National Taiwan University
**Terry Cojean**, Karlsruhe Institute of Technology
**Goran Flegar**, Universitat Jaume I
**Fritz Göbel**, Karlsruhe Institute of Technology
**Thomas Grützmacher**, Karlsruhe Institute of Technology
**Pratik Nayak**, Karlsruhe Institute of Technology
**Tobias Ribizel**, Karlsruhe Institute of Technology

> **Yuhsiang Tsai**, National Taiwan University

Reviewer 3: these two spaces can not be removed.
Otherwise, they will be in the same line
